src/main/java/graphql/GraphQL.java

```diff
@@ -25,18 +25,24 @@
 
 
     private final GraphQLSchema graphQLSchema;
-    private final ExecutionStrategy executionStrategy;
+    private final ExecutionStrategy queryStrategy;
+    private final ExecutionStrategy mutationStrategy;
 
     private static final Logger log = LoggerFactory.getLogger(GraphQL.class);
 
     public GraphQL(GraphQLSchema graphQLSchema) {
-        this(graphQLSchema, null);
+        this(graphQLSchema, null, null);
     }
 
 
-    public GraphQL(GraphQLSchema graphQLSchema, ExecutionStrategy executionStrategy) {
+    public GraphQL(GraphQLSchema graphQLSchema, ExecutionStrategy queryStrategy) {
```

Comment:
I think the cod should have a Builder of GraphQL objects so it can be extended more easily over time.

    GraphQL.newGraph()
      .schema(mySchema)
      .executionStategy(s1)
      .mutationStrategy(ms)
      .build();

I have a PR in that adds an  "Instrumentation SPI" and this is YET another example adding extra aspects to the GrapQL object